WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -560,7 +573,22 @@ private void LogCoinJoinSummary(ImmutableArray<AliceClient> registeredAliceClien
 			consolidationMode ? MaxInputsRegistrableByWallet : GetInputTarget(allowedNonPrivateCoins.Count, privateCoins.Length, rnd));
 
 		// Let's allow only a inputCount - 1 private coins to play.
-		var allowedPrivateCoins = AnonScoreBiasedShuffle(privateCoins).Take(inputCount - 1);
+		var biasSuffledPrivateCoins = AnonScoreBiasedShuffle(privateCoins).ToArray();
+		IEnumerable<SmartCoin> allowedPrivateCoins;
+		lock (SecondLargestDenomOutputLock)
```

Comment:
The lock is not required, I would just pass the SecondLargestDenomOutput as an optional parameter of SelectCoinsForRound. The method will have a local copy so it will be consistent.

Reply:
You can remove static on SecondLargestDenomOutput if you pass it as a parameter.

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/8757/commits/a2ce4d7836f2c3d8197def550b43636d04f520b2